service-base/src/main/java/org/eclipse/hono/service/management/tenant/AbstractTenantManagementHttpEndpoint.java

```diff
@@ -0,0 +1,337 @@
+     * @param vertx The Vertx instance to use.
+     * @throws NullPointerException if vertx is {@code null};
+     */
+    @Autowired
+    public AbstractTenantManagementHttpEndpoint(final Vertx vertx) {
+        super(Objects.requireNonNull(vertx));
+    }
+
+    /**
+     * Returns an empty String as this implementation does not use event bus.
+     */
+    @Override
+    protected String getEventBusAddress() {
+        return "";
+    }
+
+    @Override
+    public String getName() {
+        return TENANT_MANAGEMENT_ENDPOINT_NAME;
+    }
+
+    @Override
+    public void addRoutes(final Router router) {
+
+        final String path = String.format("/%s", getName());
+        final String pathWithTenant = String.format("/%s/:%s", getName(), PARAM_TENANT_ID);
+
+        // Add CORS handler
+        router.route(path).handler(createCorsHandler(config.getCorsAllowedOrigin(), EnumSet.of(HttpMethod.POST)));
+        router.route(pathWithTenant).handler(createDefaultCorsHandler(config.getCorsAllowedOrigin()));
+
+        final BodyHandler bodyHandler = BodyHandler.create();
+        bodyHandler.setBodyLimit(config.getMaxPayloadSize());
+
+        // ADD tenant with auto-generated ID
+        router.post(path).handler(bodyHandler);
+        router.post(path).handler(this::extractOptionalJsonPayload);
+        router.post(path).handler(this::createTenant);
+
+        // ADD tenant
+        router.post(pathWithTenant).handler(bodyHandler);
+        router.post(pathWithTenant).handler(this::extractOptionalJsonPayload);
+        router.post(pathWithTenant).handler(this::updatePayloadWithTenantId);
+        router.post(pathWithTenant).handler(this::createTenant);
+
+        // GET tenant
+        router.get(pathWithTenant).handler(this::getTenant);
+
+        // UPDATE tenant
+        router.put(pathWithTenant).handler(bodyHandler);
+        router.put(pathWithTenant).handler(this::extractRequiredJsonPayload);
+        router.put(pathWithTenant).handler(this::extractIfMatchVersionParam);
+        router.put(pathWithTenant).handler(this::updateTenant);
+
+        // REMOVE tenant
+        router.delete(pathWithTenant).handler(this::extractIfMatchVersionParam);
+        router.delete(pathWithTenant).handler(this::deleteTenant);
+    }
+
+    /**
+     * The service to forward requests to.
+     *
+     * @return The service to bind to, must never return {@code null}.
+     */
+    protected abstract TenantManagementService getService();
+
+    /**
+     * Check that the tenantId value is not blank then
+     * update the payload (that was put to the RoutingContext ctx with the
+     * key {@link #KEY_REQUEST_BODY}) with the tenant value retrieved from the RoutingContext.
+     * The tenantId value is associated with the key {@link RegistryManagementConstants#FIELD_PAYLOAD_TENANT_ID}.
+     *
+     * @param ctx The routing context to retrieve the JSON request body from.
+     */
+    protected void updatePayloadWithTenantId(final RoutingContext ctx) {
```

Comment:
Maybe this can moved to `AbstractHttpEndpoint` as the other endpoints do the same check ?

Reply:
I was putting those in `Util` class so far, but as @sophokles73 suggested we can introduce a new abstract class with all these methods, to keep `AbstractHttpEndpoint` free of device registry abstractions.